### PR TITLE
feat: BSO validation rule 

### PR DESCRIPTION
**Pull request by Warren-Pitterson**
## Description

BSO validation rule added 

## Context

Ruler 54 added 

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x ] New feature (non-breaking change which adds functionality)
- [x ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x ] I have followed the code style of the project
- [x ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: .vscode/tasks.json
[the remark this answers is not in the document]
Comment (Warren-Pitterson) [reply .vscode/tasks.json:1]: No they are not, This is my own tasks for windows. It's been reverted

**Review by andrewmorris43 — APPROVED** [review]

File: application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs [hunk 1]
```diff
@@ -28,6 +24,8 @@ public DbLookupValidationBreastScreening(IDbConnection IdbConnection)
     /// <returns>bool, whether or not the GP practice code exists in the DB.<returns>
     public bool ValidatePrimaryCareProvider(string primaryCareProvider)
     {
+        if (string.IsNullOrWhiteSpace(primaryCareProvider)) return false;
```
Comment (will-larkin-nhs) [inline application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs:27]: this logic should be in the rule

File: tests/ScreeningValidationServiceTests/LookupValidation/LookupValidationTests.cs [hunk 2]
```diff
@@ -599,6 +598,63 @@ public async Task Run_DeceasedParticipant_DoesNotCreateException(string recordTy
...
+            Times.Once());
+    }
+
+    [TestMethod]
+    [DataRow("RDI", "ValidPCP", "AL1 1BB", Actions.Amended)]
+    [DataRow("RDR", "", "AL3 0AX", Actions.Amended)]
+    [DataRow("ABC", "ZZZPCP", "", Actions.Amended)]
+    [DataRow("RPR", "", "", Actions.New)]
+
+    public async Task Run_ValidateBsoCodeRulePasses_NoExceptionIsRaised(string reasonForRemoval, string primaryCareProvider, string postcode, string recordType)
```
Comment (will-larkin-nhs) [inline tests/ScreeningValidationServiceTests/LookupValidation/LookupValidationTests.cs:633]: The scenario part of the name should not be that the rule has failed but what is causing the rule to fail (i.e what is wrong with the record)

Comment (Warren-Pitterson) [reply tests/ScreeningValidationServiceTests/LookupValidation/LookupValidationTests.cs]: Changed

File: application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs (hunk 1 above)
Comment (Warren-Pitterson) [reply application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs]: Removed

**Review by will-larkin-nhs — APPROVED** [review]

**Review by andrewmorris43 — APPROVED** [review]

**Pull request merged**